### PR TITLE
Gender enum in class Person and its validation in PESEL added

### PR DESCRIPTION
**Pull request by Aleksiiej**

### CODE REVIEW

File: src/app/person/person.cpp
Comment (fbtom) [inline src/app/person/person.cpp:64]: validatePesel(const std::string& pesel) 
is the same but without validateGender, please keep one definition of validation

File: src/app/person/person.cpp
Comment (fbtom) [inline src/app/person/person.cpp:9]: Before setting we should perform validation here. Please add validation part and respectively to the output set or not set pesel. 
I suggest using exception here.

File: src/app/person/person.cpp
Comment (fbtom) [inline src/app/person/person.cpp:134]: I think we should perform validation on pass-by value and then set it in the object. 
Please use parameter instead value from the object

File: src/app/person/person.hpp
Comment (fbtom) [inline src/app/person/person.hpp:34]: I suggest moving validation methods to private and use them properly within object.
`setPesel(pesel)
{
if(validatePesel(pesel))
{
 pesel_=pesel;
}
else {
throw exception;
}
}`

File: src/app/person/person.cpp
Comment (fbtom) [reply src/app/person/person.cpp:9]: `setPesel(pesel) { if(validatePesel(pesel)) { pesel_=pesel; } else { throw exception; } }`

File: src/tests/ut/clinic/clinic_test.cpp
Comment (fbtom) [inline src/tests/ut/clinic/clinic_test.cpp:10]: By the test we can see that validation is not performed on creation of the object. Please correct that. 
I know there's quite a lot of tests and you might need to find a few pesels to satisfy them, but it would show that our validation works as intented

File: src/app/person/person.hpp
Comment (fbtom) [inline src/app/person/person.hpp:10]: Please provide also function "toString" which will translate Gender enum to string. 
You can find toString example in alergen.cpp

**Review by fbtom — CHANGES_REQUESTED** [review]

File: src/app/person/person.cpp
Comment (mBialczak) [reply src/app/person/person.cpp:134]: I agree. In general all validation should happen before setting pesel. We don't want to set incorrect value and only then realize that it was wrong.

**Comment by ziobron** [comment]
Thanks @mBialczak for Code Review.
🏅 1 XP granted

File: src/app/person/person.cpp
Comment (fbtom) [reply src/app/person/person.cpp:58]: I agree with naming convention "hasX".
Regarding the other part you might want to get more into detail of [Boolean Algebra](https://en.wikipedia.org/wiki/Boolean_algebra_(structure)) topic

File: src/app/person/person.cpp
Comment (Aleksiiej) [reply src/app/person/person.cpp:9]: Done

File: src/app/person/person.cpp
Comment (Aleksiiej) [reply src/app/person/person.cpp:134]: Done

File: src/app/person/person.hpp
Comment (Aleksiiej) [reply src/app/person/person.hpp:34]: Done

File: src/tests/ut/clinic/clinic_test.cpp
Comment (Aleksiiej) [reply src/tests/ut/clinic/clinic_test.cpp:10]: Done

File: src/app/person/person.hpp
Comment (Aleksiiej) [reply src/app/person/person.hpp:10]: Done

File: src/app/doctor/doctor.cpp
Comment (Aleksiiej) [reply src/app/doctor/doctor.cpp:6]: Done

File: src/app/person/person.cpp
Comment (Aleksiiej) [reply src/app/person/person.cpp]: done

**Review by fbtom — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

File: src/app/person/person.cpp
Comment (mBialczak) [inline src/app/person/person.cpp:8]: I saw that you have corrected most occurences of "Gender&", so I feel being a facist for mentioning missing some spots:
1.(here).
2. Other places
./app/receptionist/receptionist.hpp:9:
./app/receptionist/receptionist.hpp:16:     
./app/receptionist/receptionist.cpp:5:                      
./app/receptionist/receptionist.cpp:21:           
./app/patient/patient.cpp:4:
./app/patient/patient.cpp:64:                      
./app/patient/patient.hpp:16:    
./app/patient/patient.hpp:31:                         
./app/person/person.hpp:21:
./app/person/person.cpp:8:
./app/doctor/doctor.hpp:11:
./app/doctor/doctor.hpp:19:                    
./app/doctor/doctor.cpp:6:
./app/doctor/doctor.cpp:27:

Sorry for me being pedantic :(

**Review by mBialczak — COMMENTED** [review]
I think, that this PR is ok and generally can be submitted to main.
Most of my comments are really just minor remarks or things to read and mark as "resolved".

I can see, that you have FBToms approval, so my review can also be considered as approval (as just one is officially required).
So you can resolve my conversations right away and merge to main if branch allows it (if there are no conflicts) or change anything if you want and treat this review as OK-eish without need to wait for mine approval as well to speed things up.

**Comment by ziobron** [comment]
Thanks @mBialczak for Code Review.
🏅 1 XP granted

File: src/app/person/person.cpp
Comment (Aleksiiej) [reply src/app/person/person.cpp:53]: Done

File: src/app/person/person.cpp
Comment (Aleksiiej) [reply src/app/person/person.cpp:60]: Done

File: src/app/person/person.cpp
Comment (Aleksiiej) [reply src/app/person/person.cpp]: Done

**Review by mBialczak — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @mBialczak for Code Review.
🏅 1 XP granted

**Review by BaqWin — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

**Pull request merged**
